src/btree/row_modify.c

```diff
@@ -399,7 +402,7 @@ __wt_update_obsolete_check(
     if (count > 20 && page->modify != NULL) {
         page->modify->obsolete_check_txn = txn_global->last_running;
         if (txn_global->has_pinned_timestamp)
-            page->modify->obsolete_check_timestamp = txn_global->pinned_timestamp;
+            page->modify->obsolete_check_timestamp = WT_MAX(txn_global->pinned_timestamp, min_ts);
```

Comment:
Curious why we are taking taking max of the two values? More comments will be appreciated.

Reply:
The idea is that it is potentially possible that pinned timestamp is greater than the minimum timestamp that we find in the update list. This case can only happen when finding an obsolete update but we don't successfully complete this if condition `if (first != NULL && (next = first->next) != NULL && __wt_atomic_cas_ptr(&first->next, next, NULL))`. 

Otherwise if that was not the case, this means that we always taking the minimum timestamp of any update list. I wonder if we want this to become `page->modify->obsolete_check_timestamp = WT_MIN(page->modify->obsolete_check_timestamp , min_ts);`. Let me know what your thoughts are here.

Reply:
Double checked the optimisation of `page->modify->obsolete_check_timestamp = WT_MIN(page->modify->obsolete_check_timestamp , min_ts)`, and found that it might not possible to achieve this effect. This is because the system's timestamp is always increasing, and trying to find the minimum of the timestamps, will lock the `page->modify->obsolete_check_timestamp` to the original value.

Reply:
I believe this is correct because if `min_ts` is greater than the `pinned_timestamp`, we won't be able to mark updates obsolete until the `pinned_timestamp` becomes greater. Can you reword this as a comment for future reference?

Reply:
I have added the comment when finding the minimum timestamp in the update list. I believe that should be suffice. Let me know if you think there is something additional we should add here.

Reply:
Should be okay I believe.